### DISCUSSION TITLE
Which voltage rating of VCC to GND capacitor?

### DISCUSSION THREAD

**Discussion by SquashSquishSquaw** [SpenceKonde/megaTinyCore#491] [Q&A]
Hi,

a capacitor is required between the MCU's VCC and GND pins:

https://github.com/SpenceKonde/megaTinyCore/blob/master/Wiring.md

I assume the purpose is to remove voltage spikes from the PSU and/or to smooth voltage. Could someone please tell me what would be a useful voltage rating for this capacitor?

Many thanks
Jenny

**Comment by mechatroniks-git** [comment]
If you are running the mcu at 5V, it should be at least 6.3V, but I like to use X7R 50V rated parts, as they are commonly available cheap. https://www.digikey.com/en/products/detail/kemet/C0805C104M5RACTU/411168. 25V is ok, 16V fine too. Anything lower, then I'd want to know more about your voltage regulation circuit or power source before recommending. 

I also like to add in a 1uF in parallel to the .1uF (can be 16V or 25V) for a bit of extra stability.

**Reply by SpenceKonde** [reply]
Yeah - it just has to be rated for at more than the applied voltage. 0.1uF caps are commonly 25-50V, even small smd ones. But for larger value capacitors, the voltage ratings get lower, particularly for physically smaller capacitors. As noted on that page, I recommend some sort of board level decoupling - 1uF minimum, and I try to use 4.7uF for most of my designs. For tantalum caps, you;re supposed to derate the voltage, owing to their reputation for "exploding" (though I've only managed to make tants smoulder - I was very disappointed) - but who uses those anymore, what with how far ceramic caps have come?

Note that the Dx-series parts have more specific requirements *which microchiop has changed twice since they released the parts*! First they said 0.1uf, then realized that some people were connecting Dx chips to supplies with lots of caps and short wires (think plugging a board with no board-level decoupling into a USB port with a pile of caps on behind it) and the voltage was ramping up so fast it exceeded the specified maximum. So the next revision of the datasheet said to use 1uF caps instead. But now they backed away from that and went back to 0.1uF as primary decoupling cap, but said you need a 1uF in addition to that if the board experiences rapid changes in Vcc. It would appear that they assumed good electrical designmanship was more common than they thought, and that they wouldn't have people building boards with no capacitance on the board except for the decoupling caps.... One really ought to always have at least a few uF of "board level decoupling" to counteract the inductance of the wires to the power supply and stuff. 

But on the Dx-series, it's not just a good idea, but critical because they don't like super-fast-rising input voltages (I think this is unique to the improved core of the Dx-series - it internally generates a like 1.6-1.8v, something in that area, that the core itself runs on.  that improved core is why the Dx-series can run at their full speed between 1.8v and 5.5v. Apparently the regulator that generates that, though, can overshoot if you go from 1.8v to 5v within 13 microseconds which can damage the chip :-P
That epic new core design is also why, equipped with an external clock, the top bin (ie, extended temperature range, E-spec) can, at least in some specimens, run at a mind-blowing 48 MHz:-P.... The AVR Dx-series has performance improvements **even more dramatic** than the (already very dramatic, esp. at low voltage) improvements advertised, provided you're willing to act with flagrant disregard for the manufacturer specs and your device does not need to run at elevated temperatures. That "48 MHz" PLL with "3x multiplication"? It's got a hidden 4x option too. And the internal oscillator can be cranked to 32, not just the 24 the datasheet says is maximum, just by increasing the value of that bitfield,. The combination can be combined  and, unbelievably (again, at room temperature),, this 128 MHz PLL clock can be used as the source for TCD0 without problems. (when I tried rthe same with 40 MHz crystal on an I-spec AVR128DB64, though, the timer output was mad glitchy (sheesh, what a shoddy product! It can't even do three timers their spec'ed maximum PLL output with a system clock of 1 and  2/3 times the rating? Terrible!!)

**Reply by SquashSquishSquaw** [reply]
Many thanks guys, for your very helpful answers, much appreciated!

For the moment I'm only learning how to make use of MCUs, so I'll be running the ATtiny 1624 on a board powered by two NiMH cells. I guess voltage won't go up too high that way, but later on I might be using some of these cheap Ebay breadboard PSUs, so I'll definitely go for your recommendations and use 4.7 uF, 25 V if possible.

> I also like to add in a 1uF in parallel to the .1uF (can be 16V or 25V) for a bit of extra stability.

Doesn't this just add to the 0.1 uF? Is there a noticeable difference between using 2 capacitors in parallel (0.1 uF + 1 uF), as opposed to using only one 1 uF capacitor?

> For tantalum caps, you;re supposed to derate the voltage, owing to their reputation for "exploding" (though I've only managed to make tants smoulder - I was very disappointed)

That sounds like you've had great fun when experimenting :-D

It's really amazing how far these tiny MCUs can be pushed!

**Reply by mechatroniks-git** [reply]
Smaller caps can filter faster frequencies. So .1uf is handling sharp noise spikes, the 1 or 4.7uf can't respond fast enough, but are good for slower spikes, like voltage droop.

**Reply by SpenceKonde** [reply]
Someone clearly didn't read what I said the first time around, I said I **couldn't** get them to explode! If I had, now that would have been fun. When  I try to get flames out of my electronics, the damned things just give a trace of smoke at best. Now. when spares aren't on hand, though, the chance of things igniting when I want combustion to be absolutely avoided, the chanvce of dramatic, catastrophic failures goes way up. There's nothing quite like that split second where you turn it on, you see the red glow of the power light, and as smoke pours forth, you remember the power light is green, and and that's a wire/trace glowing red hot... Has happened to me twice. Once on my desktop computer about a decade and a half ago. Particularly unpleasant because I'd just put it on a UPS, dso the powoer didn't shut off when I pulled out the power strip. That one turned out to work totally fine afterweards. I let the smoke clear, realized that, I'd pluged in the power wire for an old floppy drive one pin off, so ground and 12v were shorted, unplugged that, cut off  what was left of that branch of wire from the PSU, plughged in a different power connector, correctly, and all was well. 

The other time it was one of my first custom PCBs... because... you know how there are those conductive pins that poke through a through-hole circuit board? You know what ELSE is conductive? Aluminum,. commonly used as structural material when buiilding things. And - I don't remember precisely why, but that board also had the DC ground on one pin, and the positive rail of an 8S3P LiPo stack on another..... I'd set it down , instead of properly mounting it on the standoffs. that red-hot trace may well have prevented things from going considerably worse that day. by acting as a fuse. I installed some common sense overcurrent protection after that.... . Over the next few weeks I also  I also managed to blow up two or three of those garbage LM25-whatevber buck converters on the same project (those things, they *really* don't deal well with shorts, The IC pops open  with a snap noise and flash of flame. They're also generally terribad as far as buck converters go nowadays Ever since DC-DC converters with synchronous rectification were ALSO cheaper than a latte, there has been no place for the veneranble LM2596 other than the dumpster (awful parts, don't ever use them for anything, their only redeeming quality was ever the fact that the chinese buck converters are midbogglingly cheap. Also, don't go shorting out LiPo batteries. They're one of the few technological hazardsI'm not-cavalier about these days.... )

And yeah, I don't understand *why* thats true of capacitors, but I've seen the same diagrams and the heard the same explanations enough times that I accept it; that;'s why MCUs always spec a low value decoupling capacitor (right next to the chip, one per pair of power pins - there are only 2 things in you often encounter in ArduinoLand where a few inches of wire  or trace length is enough tyo make stuff not work (and they are super f'ing common): decoupling caps crystals, as well as what I referred to as "board level" decoupling. above.... I was always taught that you want a nice big cap on every bnoard! But I see a lot of people posting on forums with designs running with barely any board level decoupling so often on. And apparently the same has been happening to professional engineers, based on Dx-seriess datasheet thrash; yuou don't change your documented capacitor size 
requirements by an order of magnitude on a product you''vr been shipping for lijke  half a tyear, then , *then change it back less than a year later* unless serious customers are pissed over it. I just hope the oprofessional engineers building prototypes with  so little board-level decoupling that they were burning out MCU's aren't the same ones who chose the modern AVR line because of it['s FuSa Functional Safety certification for use in life-safety-critical devices.... use it because they're Functional Safety Certified for life safety critical devices...
Random hobby folks in Arduino Land are supposed to make mistakes like that. I'm less forgibving of, say, the guyt designing, say, the controller f my airbags.....

**Reply by SquashSquishSquaw** [reply]
> I also like to add in a 1uF in parallel to the .1uF (can be 16V or 25V) for a bit of extra stability.

> Smaller caps can filter faster frequencies. So .1uf is handling sharp noise spikes, the 1 or 4.7uf can't respond fast enough, but are good for slower spikes, like voltage droop.

Ok, thank you!

**Reply by SquashSquishSquaw** [reply]
Thanks for all the interesting information, Spence! Reminds me to stay careful with the things that I'm doing.

> you see the red glow of the power light, and as smoke pours forth, you remember the power light is green

:-D

> Someone clearly didn't read what I said the first time around, I said I **couldn't** get them to explode! If I had, now that would have been fun.

Haha, no, I did understand that, and I was extrapolating that most likely you did succeed in the end, just with some other test objects. :-) For Li-Ion batteries I do have a hell of a lot of respect, I've been using them in rc planes and I'm happy that none of them went kaboom so far!

Speaking of fireworks: In my cellar I have this huge, bigtower-sized RF capacitor which came with an even larger, sorted out device for dielectric melting of plastics (it operates at roughly 30 MHz, I believe). The capacitor is rated 15 uF at 15 kV, and since it's a bit large to use as a decoupling capacitor for ATtinys, I'm wondering whether it might be put to some other good use. :-D

I still don't get this 0.1 uF -> 1 uF -> back to 0.1 uF story. In lots of videos which I've seen, people wire ATtinys up on a breadboard without any decoupling capacitor at all, which surprises me every time. I was under the impression that using a rather large value would be best, and even considered to use 4.7 uF capacitors. But how do I have to interpret this datasheet strategy, which now again recommends 0.1 uF instead of 1 uF? What's the disadvantage of using 1 uF, and why did they revert that?
